### PR TITLE
Make Contact Info page Email or Phone optional PR2185

### LINKED ISSUE #2185: The fields on contact Info page page should be optional
## Summary

The fields (name, email and phone) in contact info page should be optional. If the email or phone is entered, they should be validated.

## Steps to reproduce

1. Start a report
2. At contact info page, enter (name and email) or (name and phone)

Actual result: The error will be displayed 
Expected result: The error should not be displayed and the entered field should be validated

## Unresolved questions

> Are there any related issues you consider out of scope for this issue that could be addressed in the future?

### PR DESCRIPTION
**Pull request by dengcn**
Fixes #2185 

# Description
After team discussion, following rule will be applied to 3 key fields (Name, Email and Phone) in Contact Info page

Only in following 3 cases where good validated value in following will get page validation pass, otherwise, warning will show.
1) Name + email
2) Name + phone
3) Name  + email + phone 

> Please include a summary of the change.
Add back phone and email validation and enable above logic in ContactInfoForm handling

# Any new packages installed?
No
> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [X ] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

**Comment by justinr86** [comment]
From a UI/UX perspective it is very confusing to see three fields marked **Required** that are not actually required. If we are going to implement this logic we need to explain to the user what is expected.

**Comment by justinr86** [comment]
I was told to remove the logic for phone number validation because we need to accept international phone numbers. Is this no longer a requirement?

**Comment by justinr86** [comment]
If we are validating phone number we should not display the help text as the error message. We should use different text or remove it all together.
phone

**Comment by dengcn** [comment]
We need to validate the 10 digits as before

**Comment by dengcn** [comment]
Yes, we need to discuss what information needs to present to user based on this validation rule.

**Comment by justinr86** [comment]
Regarding what validation logic and/or messaging to the user we should defer to @ngosset and @stephaniemgauthier

**Comment by justinr86** [comment]
This PR directly conflicts with #2200 opened by @ying-chen-cds.

**Comment by dengcn** [comment]
Based on team discussion on Aug 17, team decided to add following front component and logics
1) Add checkbox to let user to skip "name, email, phone), once checked, validation will pass no matter there is anything in (name, email, phone) or not. In the meantime, a message will pop up below to explain 
2)If this checkout is not checked, validation applies that name is required, while either valid email or valid phone can get validation pass to move to next page
PR2185_SkipContactInfoCheckBox

Code had been pushed, all tests are pass.

File: f2/C:\logs/.464a4ad0d61d053bc33259114937d4b71f7ad97f-audit.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+    "keep": {
```
Comment (justinr86) [inline f2/C:\logs/.464a4ad0d61d053bc33259114937d4b71f7ad97f-audit.json:2]: This file should not be included in the commit.

**Comment by dengcn** [comment]
Since there are many unwanted content pop in during numeric times of merge, rebase, create a new branch and review for original PR2185. This one is closed

https://github.com/cds-snc/report-a-cybercrime/pull/2234
